python/cudf/cudf/core/internals/where.py

```diff
@@ -0,0 +1,375 @@
+import warnings
+
+import numpy as np
+import pandas as pd
+
+import cudf
+
+
+def _normalize_scalars(col, other):
+    """
+    Try to normalizes scalar values as per col dtype
+    """
+    if (
+        other is not None
+        and (isinstance(other, float) and not np.isnan(other))
+    ) and (col.dtype.type(other) != other):
+        raise TypeError(
+            f"Cannot safely cast non-equivalent "
+            f"{type(other).__name__} to {col.dtype.name}"
+        )
+
+    return (
+        col.dtype.type(other)
+        if (
+            other is not None
```

Comment:
Shouldn't need to check if `other is not None` here since you do an `isinstance(other, float)` below.